profile-specific-challenges/frontend/components/idioms.jsx

```diff
@@ -177,20 +202,35 @@ export function CrudeStateManagement() {
 
   return (
     <form onSubmit={handleSubmit}>
-      <input value={name} name="name" type="text" onChange={setName} />
-      <input value={age} name="age" type="number" onChange={setAge} />
+      <input
+        value={name}
+        name="name"
+        type="text"
+        onChange={(e) => setName(e.target.value)}
+      />
+      <input
+        value={age}
+        name="age"
+        type="number"
+        onChange={(e) => setAge(e.target.value)}
+      />
       <input
         value={location}
         name="location"
         type="text"
-        onChange={setLocation}
+        onChange={(e) => setLocation(e.target.value)}
+      />
+      <input
+        value={email}
+        name="email"
+        type="email"
+        onChange={(e) => setEmail(e.target.value)}
       />
-      <input value={email} name="email" type="email" onChange={setEmail} />
       <input
         value={password}
         name="password"
         type="password"
-        onChange={setPassword}
+        onChange={(e) => setPassword(e.target.value)}
```

Comment:
Same as above. This was missing in the original definition, but the actual solution is more elegant. Tip: use the name attribute of the input.